src/openvic-simulation/map/Province.cpp

```diff
@@ -116,57 +124,241 @@ void Province::tick(Date today) {
 	}
 }
 
-Province::adjacency_t::adjacency_t(Province const* province, distance_t distance, flags_t flags)
-	: province { province }, distance { distance }, flags { flags } {
-	assert(province != nullptr);
+Province::adjacency_t::adjacency_t(
+	Province const* new_to, distance_t new_distance, type_t new_type, Province const* new_through, data_t new_data
+) : to { new_to }, distance { new_distance }, type { new_type }, through { new_through }, data { new_data } {}
+
+std::string_view Province::adjacency_t::get_type_name(type_t type) {
+	switch (type) {
+	case type_t::LAND: return "Land";
+	case type_t::WATER: return "Water";
+	case type_t::COASTAL: return "Coastal";
+	case type_t::IMPASSABLE: return "Impassable";
+	case type_t::STRAIT: return "Strait";
+	case type_t::CANAL: return "Canal";
+	default: return "Invalid Adjacency Type";
+	}
+}
+
+Province::adjacency_t* Province::get_adjacency_to(Province const* province) {
+	const std::vector<adjacency_t>::iterator it = std::find_if(adjacencies.begin(), adjacencies.end(),
+		[province](adjacency_t const& adj) -> bool { return adj.get_to() == province; }
+	);
+	if (it != adjacencies.end()) {
+		return &*it;
+	} else {
+		return nullptr;
+	}
+}
+
+Province::adjacency_t const* Province::get_adjacency_to(Province const* province) const {
```

Comment:
How does this differ from the other get_adjacency_to?

Reply:
Public const version (for getting adjacency info e.g. for pathing) vs private non-const version (for updating adjacencies when loading the extra ones from `adjacencies.csv`, e.g. making an adjacency impassable).